src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/peripheral/RSBridgePeripheral.java

```diff
@@ -0,0 +1,922 @@
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult craftItem(IComputerAccess computer, IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        Pair<ItemFilter, String> filter = ItemFilter.parse(arguments.getTable(0));
+        if (filter.rightPresent())
+            return MethodResult.of(null, filter.getRight());
+
+        RSCraftJob job = new RSCraftJob(computer, getLevel(), filter.getLeft().getCount(), ItemResource.ofItemStack(filter.getLeft().toItemStack()), getNetwork().getComponent(AutocraftingNetworkComponent.class));
+        bridge.addJob(job);
+        return MethodResult.of(job);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult craftFluid(IComputerAccess computer, IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        Pair<FluidFilter, String> filter = FluidFilter.parse(arguments.getTable(0));
+        if (filter.rightPresent())
+            return MethodResult.of(null, filter.getRight());
+
+        RSCraftJob job = new RSCraftJob(computer, getLevel(), filter.getLeft().getCount(), VariantUtil.ofFluidStack(filter.getLeft().toFluidStack()), getNetwork().getComponent(AutocraftingNetworkComponent.class));
+        bridge.addJob(job);
+        return MethodResult.of(job);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult craftChemical(IComputerAccess computer, IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        if (!APAddons.mekanismLoaded || !APAddons.refinedStorageMekanismLoaded)
+            return MethodResult.of(Collections.emptyList());
+
+        Pair<ChemicalFilter, String> filter = ChemicalFilter.parse(arguments.getTable(0));
+        if (filter.rightPresent())
+            return MethodResult.of(null, filter.getRight());
+
+        //RSCraftJob job = new RSCraftJob(computer, getLevel(), filter.getLeft().getCount(), ChemicalResource.ofChemicalStack(filter.getLeft().toChemicalStack()), getNetwork().getComponent(AutocraftingNetworkComponent.class));
+        //bridge.addJob(job);
+        //return MethodResult.of(job);
+        return MethodResult.of(0);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult getCraftingTasks() {
+        if (!isAvailable())
+            return notConnected();
+
+        return MethodResult.of(RsApi.getCraftingTasks(bridge));
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult getCraftingTask(int id) {
+        if (!isAvailable())
+            return notConnected();
+
+        RSCraftJob foundJob = null;
+
+        for (RSCraftJob job : bridge.getJobs()) {
+            if (job.getId() == id) {
+                foundJob = job;
```

Comment:
can break here

actually what I will do is directly return at here, and then return a empty methodresult after the for

Reply:
ahh yeah
totally no clue why I did it like that